### PR TITLE
feat: Add support for getting api key from envvars

### PR DESCRIPTION
**Pull request by mcataford**
## Description

This allows us to use the Jenkins environment variables to set an API key, no more hard coding, ever!

## DevQA
- Go to `Manage Jenkins` then `Configure system`
- Set the API key as an environment variable (see under `Global properties`)
image
- Run a build and verify that the printed status code is `202`, signalling that the payload was accepted by DD.
- A 403 is what we would see with a missing API key, which you can also test. The system should be resilient to the absence of a key and should just 403 when sending to DD.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/statspublisher/JenkinsRunPayload.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
   private String upstreamCause;
   private String result;
 
-  private JenkinsRunPayload() {}
+  public JenkinsRunPayload() {}
```
Comment (mcataford) [inline src/main/java/io/jenkins/plugins/statspublisher/JenkinsRunPayload.java]:
```suggestion
  private JenkinsRunPayload() {}
```

File: src/main/java/io/jenkins/plugins/statspublisher/JenkinsRunListener.java [hunk 2]
```diff
@@ -49,13 +48,24 @@ public final void onCompleted(Run run, @Nonnull TaskListener listener) {
...
-      int returnCode = DataDogRequestHandler.sendTimeseriesPayload("key", json);
+      String apiKey = getDataDogApiKey(run, listener);
+      int returnCode = DataDogRequestHandler.sendTimeseriesPayload(apiKey, json);
       LOG.info("" + returnCode);
     } catch (Exception e) {
       LOG.info(e.toString());
     }
   }
 
+  private String getDataDogApiKey(Run run, TaskListener listener) {
```
Comment (leevictor81) [inline src/main/java/io/jenkins/plugins/statspublisher/JenkinsRunListener.java]: actually sorry shouldn't we put this in DataDogRequestHandler instead

Comment (mcataford) [reply src/main/java/io/jenkins/plugins/statspublisher/JenkinsRunListener.java]: I was thinking about it; then we'd need to pass both the `run` and the `listener` in there, which kind of feels like bloat given that we only extract the one item.

Comment (leevictor81) [reply src/main/java/io/jenkins/plugins/statspublisher/JenkinsRunListener.java]: okay makes sense. 👍

**Pull request merged**